### PR TITLE
fixpoint util optimizations

### PR DESCRIPTION
**Pull request by kgioioso**
1) function arrays are now constant
2) define and use memory.copy and table.copy instead of my diy version
3) grow memory by bytes/page_size instead of bytes

### CODE REVIEW

File: applications/flatware/asm-flatware.wat [hunk 1]
```diff
@@ -1,5 +1,13 @@
 (module
   (import "c-flatware" "memory" (memory $flatware_mem 0))
+  (import "fixpoint_storage_0" "rw_mem_0" (memory $rw_mem_0 0))
+  (import "fixpoint_storage_0" "rw_mem_1" (memory $rw_mem_1 0))
+  (import "fixpoint_storage_0" "rw_mem_2" (memory $rw_mem_2 0))
```
Comment (kgioioso) [inline applications/flatware/asm-flatware.wat:5]: The linking and importing for flatware is a little odd. Here is what I do:
1) copy `fixpoint_storage.wasm` to `fixpoint_storage_0.wasm`
2) wasmlink `asm-flatware.wasm` and `fixpoint_storage_0.wasm` -> output as `fixpoint_storage.wasm`
3) wasmlink `fixpoint_storage.wasm` with `c-flatware.wasm`

Thus 
1) asm-flatware.wat imports things from fixpoint_storage_0 that are originally defined in fixpoint_storage.wat
2) asm-flatware.c imports things from fixpoint_storage that are originally defined in asm-flatware.wat

This seems confusing. I tried multiple other things: 
1) wasmld fixpoint_storage, asm-flatware, and the flatware c files . This breaks once we try generating `wasi-snaphort-preview` (and wasm2wat errors with error: memory index 1 out of range)
2) wasmlink fixpoint_storage with asm-flatware and separately wasmlink fixpoint_storage with c-flatware, and then wasmlink the two separate wasm files -> this errors with duplicate symbols

Any suggestions?

Comment (yhdengh) [reply applications/flatware/asm-flatware.wat:5]: wasmld does not handle multiple memories correctly, which was the main reason why we had this wasmlink. Actually, wasmlink allows you to specify the module name on command line with `-m` and `-n` option (which can be different from the actual module name). So maybe:

1.    wasm2wat asm-flatware.wasm -> output as "asm-flatware-no-storage.wasm"
2.    wasmlink c-flatware.wasm with asm-flatware-no-storage.wasm (with name asm-flatware) -> output as "c-flatware-no-storage.wasm"
3.    wasmlink c-flatware-on-storage.wasm with fixpoint_storage_0.wasm (with name fixpoint_storage) -> output as "linked_with_storage.wasm" 

Then, asm-flatware.c imports from asm-flatware, and asm-flatware.wat imports from fixpoint_storage?

Comment (kgioioso) [reply applications/flatware/asm-flatware.wat]: that's way cleaner. Thanks, Yuhan!

**Review by yhdengh — APPROVED** [review]
lgtm

**Pull request merged**